File: backend/app/schemas/checklist.py
from .common import CamelModel


class ChecklistTaskDTO(CamelModel):
    id: int
    title: str
    description: str | None
    repeat_type: str
    repeat_interval_days: int | None
    created_at: int
    last_completed_at: int | None
    is_done_current_cycle: bool


def checklist_task_to_dto(row: dict) -> ChecklistTaskDTO:
    return ChecklistTaskDTO(**row)


class ChecklistListResponse(CamelModel):
    tasks: list[ChecklistTaskDTO]


class ChecklistTaskResponse(CamelModel):
    task: ChecklistTaskDTO


class ChecklistTaskRequest(CamelModel):
    title: str
    description: str | None = None
    repeat_type: str
    repeat_interval_days: int | None = None


class SetDoneRequest(CamelModel):
    done: bool

File: frontend/src/api/checklist.ts
import { apiFetch } from './client';
import type { ChecklistTaskDTO, RepeatType } from '../types/api';

export interface ChecklistTaskInput {
  title: string;
  description: string;
  repeatType: RepeatType;
  repeatIntervalDays: number | null;
}

export function getChecklist(): Promise<{ tasks: ChecklistTaskDTO[] }> {
  return apiFetch('/api/checklist');
}

export function createTask(input: ChecklistTaskInput): Promise<{ task: ChecklistTaskDTO }> {
  return apiFetch('/api/checklist', { method: 'POST', body: JSON.stringify(input) });
}

export function updateTask(id: number, input: ChecklistTaskInput): Promise<{ task: ChecklistTaskDTO }> {
  return apiFetch(`/api/checklist/${id}`, { method: 'PATCH', body: JSON.stringify(input) });
}

export function deleteTask(id: number): Promise<void> {
  return apiFetch(`/api/checklist/${id}`, { method: 'DELETE' });
}

export function setTaskDone(id: number, done: boolean): Promise<{ task: ChecklistTaskDTO }> {
  return apiFetch(`/api/checklist/${id}/done`, { method: 'POST', body: JSON.stringify({ done }) });
}
